src/pybind/mgr/dashboard/controllers/rgw.py

```diff
@@ -129,8 +129,17 @@ def get_sync_status(self):
     @Endpoint(path='/sync-policy')
     @EndpointDoc("Get the sync policy")
     @ReadPermission
-    def get_sync_policy(self, bucket_name='', zonegroup_name=''):
+    def get_sync_policy(self, bucket_name='', zonegroup_name='', fetch_all_bucket= None):
         multisite_instance = RgwMultisite()
+        fetch_all_bucket = str_to_bool(fetch_all_bucket)
+        if fetch_all_bucket:
+            sync_policy_list = []
+            rgw_bucket = RgwBucket()
+            bucket_list = json.loads(rgw_bucket.list(stats=False))
```

Comment:
and we don't need an extra object instantiation. also a proper variable naming
```suggestion
            buckets = json.loads(RgwBucket().list(stats=False))
```